receiver/prometheusreceiver/internal/metricsutil_pdata.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal"
+
+import "go.opentelemetry.io/collector/model/pdata"
+
+type kv struct {
+	Key, Value string
+}
+
+func distPointPdata(ts pdata.Timestamp, bounds []float64, counts []uint64) *pdata.HistogramDataPoint {
+	hdp := pdata.NewHistogramDataPoint()
+	hdp.SetExplicitBounds(bounds)
+	hdp.SetBucketCounts(counts)
+	hdp.SetTimestamp(ts)
+	var sum float64
+	var count uint64
+	for i, bcount := range counts {
+		count += bcount
+		if i > 0 {
+			sum += float64(bcount) * bounds[i-1]
+		}
+	}
+	hdp.SetCount(count)
+	hdp.SetSum(sum)
+
+	return &hdp
+}
+
+func gaugeDistMetricPdata(name string, kvp []*kv, startTs pdata.Timestamp, points ...*pdata.HistogramDataPoint) *pdata.Metric {
+	hMetric := cumulativeDistMetricPdata(name, kvp, startTs, points...)
+	hMetric.Histogram().SetAggregationTemporality(pdata.MetricAggregationTemporalityDelta)
+	return hMetric
+}
+
+func cumulativeDistMetricPdata(name string, kvp []*kv, startTs pdata.Timestamp, points ...*pdata.HistogramDataPoint) *pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.SetName(name)
+	metric.SetDataType(pdata.MetricDataTypeHistogram)
+	histogram := metric.Histogram()
+	histogram.SetAggregationTemporality(pdata.MetricAggregationTemporalityCumulative)
+
+	destPointL := histogram.DataPoints()
+	// By default the AggregationTemporality is Cumulative until it'll be changed by the caller.
```

Comment:
From what I can tell this is referring to line 53

Reply:
I think this comment is either a vestige of a prior pdata API or a misunderstanding by the original author.  The default aggregation temporality appears to be unspecified.  See https://github.com/open-telemetry/opentelemetry-collector/blob/main/model/pdata/metrics.go#L256-L261